qucs/qucs/qucsdoc.cpp

```diff
@@ -71,3 +71,16 @@ QString QucsDoc::fileBase (void) {
   return fileBase (DocName);
 }
 
+// Returns a boolean flag indicating if the document has write permissions
+bool QucsDoc::isreadonly()
+{
+    if (DocName.isEmpty())
+    {//The file wasn't created yet
+     return false;
+    }
+    else
+    {
+      QFileInfo Info(DocName);
+      return !Info.isWritable();
```

Comment:
for non-existing files `isWritable()` seems to return `false` so things do not work as expected in this case:

- open a read-only schematic
- save it with a new name
- run a simulation
- a new Data Display will open but it ill be incorrectly tagged as `[READ-ONLY]`

Reply:
Reproduced. Thanks!